File: sdk/py/examples/audio/ex_audio.py
# OA_DOC_BEGIN: audio-process
import os
import sys

import oa

engine = oa.Engine()

# Asset path: OA_ASSET env var or the SDK asset directory by convention.
asset_root = os.environ.get("OA_ASSET", "asset")
audio_path = os.path.join(asset_root, "audio/oaNarration.wav")

audio = oa.audio.decode_file(engine, audio_path)

mono = oa.audio.to_mono(audio)

normalized = oa.audio.normalize(mono, -3.0)

faded = oa.audio.fade(normalized, 2400, 2400)

reverberated = oa.audio.reverb(faded, 1.5, 0.4)

var_root = os.environ.get("OA_VAR", "var")
output = os.path.join(var_root, "example/audio/oaNarrationRoom.wav")
os.makedirs(os.path.dirname(output), exist_ok=True)
oa.audio.save_wav_f32(output, reverberated)

assert reverberated.channels == 1
assert os.path.isfile(output)

print(f"Saved reverberated audio (1.5 s tail): {output}")
# OA_DOC_END: audio-process

File: sdk/rs/examples/audio/ex_audio.rs
// OA_DOC_BEGIN: audio-process
use anyhow::Context;
use oa::{Engine, Filesystem, Path, audio};

fn main() -> anyhow::Result<()> {
	let engine = Engine::new().context("create OA engine")?;

	let audio_path = Path::asset_rel("audio/oaNarration.wav");

	let output = Path::var_rel("example/audio/oaNarrationRoom.wav");
	Filesystem::create_directories(&output.parent().unwrap())?;

	let source = audio::decode_file(&engine, &audio_path)
		.with_context(|| format!("decode {}", audio_path.display()))?;
	let mono = audio::to_mono(&source)?;
	let normalized = audio::normalize(
		&mono,
		audio::NormalizeAudioConfig {
			target_db: -3.0,
			..Default::default()
		},
	)?;
	let faded = audio::fade(&normalized, 2400, 2400)?;
	let reverberated = audio::reverb(&faded, 1.5, 0.4)?;
	audio::save_wav_f32(&output, &reverberated)?;

	assert!(reverberated.channels() == 1);
	assert!(output.is_file());

	println!(
		"Saved reverberated audio (1.5 s tail): {}",
		output.display()
	);
	Ok(())
}
// OA_DOC_END: audio-process
